app/classes/custom_fields/data_types/number.rb

```diff
@@ -28,11 +31,25 @@ def coerce_value(value)
 
       private
 
+      def normalize_trailing_zeros(str)
+        str.gsub(TRAILING_ZEROS_REGEX, '\1').gsub(TRAILING_DOT_REGEX, '')
+      end
+
+      def valid_number_string?(str)
+        return false if str.empty?
```

Comment:
This check looks redundant because coerce_value return nil in case of empty string and we check this later on